app/javascript/gobierto_data/webapp/components/sets/data/SQLEditorResults.vue

```diff
@@ -127,54 +122,66 @@ export default {
       labelVisualize: I18n.t('gobierto_data.projects.visualize') || "",
       labelResetViz: I18n.t('gobierto_data.projects.resetViz') || "",
       labelModifiedVizualition: I18n.t("gobierto_data.projects.modifiedVisualization") || "",
+      labelSavedVisualization: I18n.t("gobierto_data.projects.savedVisualization") || "",
       showVisualization: false,
-      isVisualizationModified: false,
       showResetViz: false,
       showVisualize: true,
       removeLabelBtn: false,
       perspectiveChanged: false,
-      typeChart: 'hypergrid'
+      typeChart: 'hypergrid',
     };
   },
   methods: {
     onSaveEventHandler(opts) {
       // get children configuration
       const config = this.$refs.viewer.getConfig()
 
-      this.$root.$emit("storeCurrentVisualization", config, opts);
-      this.removeLabelBtn = true
-      this.isVisualizationModified = false
+      if (!this.isVizSavingPromptVisible) {
+        this.$root.$emit("isVizSavingPromptVisible", true);
+        this.$nextTick(() => this.$refs.savingDialogViz.inputFocus());
+      } else {
+        if (!this.labelValue) {
+          this.$nextTick(() => this.$refs.savingDialogViz.inputFocus())
+        } else {
+          this.$root.$emit("storeCurrentVisualization", config, opts);
+        }
+      }
+    },
+    updateVizName(value) {
+      const {
+        name: vizName
+      } = value;
+      this.labelValue = vizName
+      this.$root.$emit("eventIsVizModified", true);
     },
     resetViz() {
       const hidePerspective = "none"
 
       this.showVisualize = true
       this.perspectiveChanged = false
-      this.removeLabelBtn = false
       this.showResetViz = false
       this.typeChart = 'hypergrid'
-      this.isVisualizationModified = false
 
       this.$refs.viewer.enableDisabledPerspective(hidePerspective);
       this.$refs.viewer.setColumns();
+      this.$root.$emit('enableSavedVizButton', true)
+      this.$root.$emit('disabledSavedVizString')
     },
     showChart() {
       const showPerspective = "flex"
-
       this.showVisualization = true
       this.$refs.viewer.enableDisabledPerspective(showPerspective);
     },
     showSavingDialog() {
       this.perspectiveChanged = true
       this.showVisualize = false
       this.showResetViz = true
-      this.isVisualizationModified = true
-      this.$root.$emit('enableSavedButton')
-      this.$root.$emit("isSavingPromptVisible", true);
-    },
-    resetButtonViz() {
-      this.removeLabelBtn = false
-      this.showResetViz = true
+      this.$root.$emit('enableSavedVizButton', true)
+      if (this.isUserLogged) {
```

Comment:
For instance, what does _SQLEditorResults_ care about if the user is or not logged? It's not related

Reply:
`SQLEditorResults` contains the `SavingDialog` of Visualizations. If the user is not logged in, we'll redirect him to the login.